### PR TITLE
ceph: add RGW to embedded ceph

### PR DESCRIPTION
**Pull request by bassamtabbara**
With this PR we now build some of the packages ceph depends on directly from source code. We use cmake and externalproject_add to build the sources using a superbuild project approach. Not all
packages that are compiled into rook are built from source yet, but enough are there to work around some of the issues we were ran into with adding RGW to rook. 

Specificaly, libcurl is now built from source and removes the dependency on openldap and other unused protocols. Boost, zlib, libblkid, and libuuid are now also built from source. More packages
will be added in the future to complete this approach.

Also picked up new embedded ceph with RGW exposed.

### CODE REVIEW

**Comment by bassamtabbara** [comment]
retest this please

**Comment by bassamtabbara** [comment]
ok to test

**Comment by jbw976** [comment]
> More packages will be added in the future to complete this approach.

are we tracking this anywhere?

File: build/container/Dockerfile [hunk 1]
```diff
@@ -182,7 +113,83 @@ RUN \
...
+    DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-overwrite" install -yq --no-install-recommends \
+    libatomic-ops-dev:${a} \
+    libexpat1-dev:${a} \
+    libkeyutils-dev:${a} \
+    libsnappy-dev:${a} \
+    libssl-dev:${a} \
+    libudev-dev:${a} \
+    ; done
+
+# Some of the packages we need for ceph are either not available in yakkety or dont support multi-arch.
```
Comment (jbw976) [inline build/container/Dockerfile]: should a BUGBUG or TODO be added to the start of the comment here so lingering issues can be found by grepping the rook repo later on?

File: build/container/external/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# set the cross triple used
+set(EXTERNAL_CROSS_TRIPLE ${CROSS_TRIPLE})
+
+# set external download/cache dir
+set(EXTERNAL_INSTALL_DIR ${CMAKE_INSTALL_PREFIX})
+
+# whether to log the configure, build and install commands
+# or have them go to stdout.
+option(EXTERNAL_LOGGING "whether to log all external build output to files" ON)
```
Comment (jbw976) [inline build/container/external/CMakeLists.txt]: where does the external logging go?

File: build/container/external/packages/boost.cmake [hunk 3]
```diff
@@ -0,0 +1,62 @@
+
```
Comment (jbw976) [inline build/container/external/packages/boost.cmake]: are these external package cmake files largely untouched from the build system of their original repo?  or did you have to author a lot of new logic for them?

File: build/container/external/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+message(STATUS "External: Installing to ${EXTERNAL_INSTALL_DIR}")
+
+#
+# Build the external projects in order of dependencies
+#
+
+include(packages/zlib.cmake)
+include(packages/boost.cmake)
+include(packages/curl.cmake)
+include(packages/util-linux.cmake)
```
Comment (jbw976) [inline build/container/external/CMakeLists.txt]: why isn't cryptopp.cmake in this list too?

File: build/container/version [hunk 5]
```diff
@@ -0,0 +1 @@
+0.2.0
```
Comment (jbw976) [inline build/container/version]: smart idea to bump the minor here

**Comment by jbw976** [comment]
Also, do we have a good system in place (or ideas about a process) to track CVEs and get security fixes integrated ASAP?

File: build/container/Dockerfile (hunk 1 above)
Comment (bassamtabbara) [inline build/container/Dockerfile]: I'm working my way through all these packages. this whole section should go away

File: build/container/external/CMakeLists.txt (hunk 4 above)
Comment (bassamtabbara) [inline build/container/external/CMakeLists.txt]: its not ready yet. working on that next.

**Comment by bassamtabbara** [comment]
I'm going to write down some thoughts in a wiki page regarding security fixes and CVEs.

**Pull request merged**